qiskit_dynamics/backend/dynamics_backend.py

```diff
@@ -182,23 +196,37 @@ def set_options(self, **fields):
                         """initial_state must be either "ground_state",
                         or a Statevector or DensityMatrix instance."""
                     )
-            elif key == "meas_level" and value != 2:
-                raise QiskitError("Only meas_level == 2 is supported by DynamicsBackend.")
+            elif key == "meas_level" and value not in [1, 2]:
+                raise QiskitError("Only meas_level 1 and 2 are supported by DynamicsBackend.")
+            elif key == "meas_return" and value not in ["single", "avg"]:
+                raise QiskitError("meas_return must be either 'single' or 'avg'.")
             elif key == "max_outcome_level":
                 if (value is not None) and (not isinstance(value, int) or (value <= 0)):
                     raise QiskitError("max_outcome_level must be a positive integer or None.")
             elif key == "experiment_result_function" and not callable(value):
                 raise QiskitError("experiment_result_function must be callable.")
+            elif key == "iq_width" and (not isinstance(value, float) or (value <= 0)):
+                raise QiskitError("iq_width must be positive float.")
+            elif key == "iq_centers":
```

Comment:
This also needs to accept `value is None` right? That is a valid value for `iq_centers`. You should also check that `isinstance(iq_centers, list)`.

The error message here should be more informative and follow proper capitalization:
```
"The iq_centers option must be either None or of type List[List[List[int, int]]], where the outer most list is of length equal to the number of subsystems, and each inner list of length equal to the corresponding subsystem dimension."
```

Reply:
Here we only check if the IQ points are lists of 2 elements. Not using any info about the number of subsystems or their dimensions. So maybe for this check, we can have:
```
"The iq_centers option must be either None or of type List[List[List[int, int]]], where the innermost list is the (I, Q) pair."
```` 
And have the error message you suggested above for the consistency check.